src/ast/semantic_analyser.cpp

```diff
@@ -701,14 +748,16 @@ void SemanticAnalyser::visit(Call &call)
         }
       }
     }
-
-    call.type = CreateUInt64();
+    // Case : $a=*reg("ip"). Check provider and decide the addrspace.
+    ProbeType pt = probetype((*probe_->attach_points)[0]->provider);
+    call.type = CreatePointer(CreateUInt64(), find_addrspace(pt));
   }
   else if (call.func == "kaddr") {
     if (check_nargs(call, 1)) {
       check_arg(call, Type::string, 0, true);
     }
     call.type = CreateUInt64();
```

Comment:
```suggestion
```

Reply:
Ok. Will change the call.type to uint64 and also set the addrspace here. Needed in cases like uprobe:fn { @=sum(*(uint16*)(reg("sp")+8)); }

Reply:
Ok. Forgot to remove that.

Reply:
This might cause a regression, #1507

Reply:
Kept the same type.